pkg/annotations/parser.go

```diff
@@ -35,31 +69,100 @@ type suffixAnnotationParser struct {
+	for _, key := range keys {
+		if raw, ok := annotations[key]; ok {
+			*value = raw
+			return true
+		}
 	}
-	*value = raw
-	return true
+	return false
 }
 
-func (p *suffixAnnotationParser) ParseInt64Annotation(suffix string, value *int64, annotations ...map[string]string) (bool, error) {
-	key := p.buildAnnotationKey(suffix)
-	raw, ok := algorithm.MapFindFirst(key, annotations...)
-	if !ok {
+func (p *suffixAnnotationParser) ParseInt64Annotation(annotation string, value *int64, annotations map[string]string, opts ...ParseOption) (bool, error) {
+	raw := ""
+	if !p.ParseStringAnnotation(annotation, &raw, annotations, opts...) {
 		return false, nil
 	}
 	i, err := strconv.ParseInt(raw, 10, 64)
 	if err != nil {
-		return true, errors.Wrapf(err, "failed to parse annotation, %v: %v", key, raw)
+		return true, errors.Wrapf(err, "failed to parse annotation, %v: %v", annotation, raw)
 	}
 	*value = i
 	return true, nil
 }
 
-// buildAnnotationKey returns full annotation key based on suffix
-func (p *suffixAnnotationParser) buildAnnotationKey(suffix string) string {
-	return fmt.Sprintf("%v/%v", p.annotationPrefix, suffix)
+func (p *suffixAnnotationParser) ParseStringSliceAnnotation(annotation string, value *[]string, annotations map[string]string, opts ...ParseOption) bool {
+	raw := ""
+	if !p.ParseStringAnnotation(annotation, &raw, annotations, opts...) {
+		return false
+	}
+	result := []string{}
+	parts := strings.Split(raw, ",")
+	for _, part := range parts {
+		part = strings.TrimSpace(part)
+		if len(part) == 0 {
+			continue
+		}
+		result = append(result, part)
+	}
+	*value = result
+	return true
+}
+
+func (p *suffixAnnotationParser) ParseJSONAnnotation(annotation string, value interface{}, annotations map[string]string, opts ...ParseOption) (bool, error) {
+	raw := ""
+	if !p.ParseStringAnnotation(annotation, &raw, annotations, opts...) {
+		return false, nil
+	}
+	if err := json.Unmarshal([]byte(raw), value); err != nil {
+		return true, errors.Wrapf(err, "failed to parse annotation, %v: %v", annotation, raw)
+	}
+	return true, nil
+}
+
+func (p *suffixAnnotationParser) ParseStringMapAnnotation(annotation string, value *map[string]string, annotations map[string]string, opts ...ParseOption) bool {
+	keyValues := make(map[string]string)
+	var result []string
+	if !p.ParseStringSliceAnnotation(annotation, &result, annotations, opts...) {
+		return false
+	}
+	for _, item := range result {
+		parts := strings.Split(strings.TrimSpace(item), "=")
+		if len(parts) >= 2 {
+			keyValues[strings.TrimSpace(parts[0])] = strings.TrimSpace(parts[1])
+		} else if len(parts) == 1 && parts[0] != "" {
+			keyValues[strings.TrimSpace(parts[0])] = ""
+		}
+	}
+	if value != nil {
+		*value = keyValues
+	}
+	return true
+}
+
+// buildAnnotationKey returns list of full annotation keys based on suffix and parse options
+func (p *suffixAnnotationParser) buildAnnotationKeys(suffix string, opts ...ParseOption) []string {
+	keys := []string{}
+	exact := false
```

Comment:
the idiomatic pattern is
```
opts :=  ParseOptions{
      exact: false, (optional since the zero value is false)
      alternativePrefixes: nil (optional since the zero value is nil)
}
for _, opt := range opts {
      opt(opts)
}
if opts.exact {
     ....
} else {
     ....
}
```

Reply:
ok